notebooks/corridor/corridor_bucharest_river-valley.qmd

```diff
@@ -0,0 +1,299 @@
+
+```{r}
+valley <- load_data(data_dir, "valley", river_name)
+```
+
+Visualize input data:
+
+```{r}
+leaflet() |>
+    addTiles() |>
+    addPolylines(data = city_boundary |> get_geom_latlon(), color = "yellow", group = "city") |>
+    addPolygons(data = valley |> get_geom_latlon(), color = "orange", group = "valley") |>
+    addPolylines(data = waterway |> get_geom_latlon(), color = "blue", group = "water") |>
+    addPolygons(data = waterbody |> get_geom_latlon(), color = "cyan", group = "water") |>
+    addPolylines(data = highways |> get_geom_latlon(), color = "black", group = "highways") |>
+    addPolylines(data = railways |> get_geom_latlon(), color = "red", group = "railways") |>
+    addLayersControl(overlayGroups = c("city", "valley", "water", "highways", "railways"))
+```
+
+## 2. Street network ----
+
+We use the highway geometries to create the street network using `sfnetworks`:
+
+```{r}
+network <- highways |>
+    as_sfnetwork(directed = FALSE)
+```
+
+We perform standard cleaning tasks on the network:
+
+```{r}
+# Simplify the graph, removing loops and double-edge connections
+# https://luukvdmeer.github.io/sfnetworks/articles/sfn02_preprocess_clean.html#simplify-network
+simplify <- function(net){
+    net|>
+        activate("edges") |>
+        # reorder the edges so that the shortest is kept
+        arrange(edge_length()) |>
+        filter(!edge_is_multiple()) |>
+        filter(!edge_is_loop())
+}
+
+clean <- function(net){
+    net |>
+        # subdivide edges by adding missing nodes
+        # https://luukvdmeer.github.io/sfnetworks/articles/sfn02_preprocess_clean.html#subdivide-edges
+        convert(to_spatial_subdivision, .clean = TRUE) |>
+        # remove pseudo-nodes
+        # https://luukvdmeer.github.io/sfnetworks/articles/sfn02_preprocess_clean.html#smooth-pseudo-nodes
+        convert(to_spatial_smooth, .clean = TRUE) |>
+        # run simplification steps
+        simplify() |>
+        # keep only the main connected component of the network
+        activate("nodes") |>
+        filter(group_components() == 1)
+}
+
+network_cleaned <- clean(network)
+```
+
+Visualize cleaned network:
+
+```{r}
+leaflet() |>
+    addTiles() |>
+    addPolylines(data = network_cleaned |> get_edges() |> get_geom_latlon(), color = "black") |>
+    addCircles(data = network_cleaned |> get_nodes() |> get_geom_latlon(), color = "red")
+```
+
+
+## 3. Corridor edge delineation ----
+
+We carry out the delineation of the river corridor in few steps.
+
+First, we consider the nodes of the network within the area of interest (here defined using the bounding box of the city) and label them according to which side of the river they fall on:
+
+```{r}
+# split polygon in two fragments using the waterway geometry as dividing line
+# if more than two fragments are obtained, keep only the two largest fragments
```

Comment:
Does this mean that any islands created by the multiple branches of the river are removed?

Reply:
As discussed, yes, smaller regions like islands will be removed. But "artificial" fragments created e.g. by the river flowing close to the boundary of the polygon would be ignored.